lnwallet/commitment.go

```diff
@@ -960,3 +962,75 @@ func addHTLC(commitTx *wire.MsgTx, ourCommit bool,
 
 	return nil
 }
+
+// findOutputIndexesFromRemote finds the index of our and their outputs from
+// the remote commitment transaction. It derives the king ring to compute the
+// output scripts and compares them against the outputs inside the commitment
+// to find the match.
+// TODO(yy): add test.
+func findOutputIndexesFromRemote(revocationPreimage *chainhash.Hash,
```

Comment:
IIRC we already have something like this in the breach handling code in the contract court?